From patchwork Tue Oct  9 15:14:30 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Li Zhijian <zhijianx.li@intel.com>
X-Patchwork-Id: 10632843
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 1B68A13AA
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  9 Oct 2018 14:47:31 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 0270728DB0
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  9 Oct 2018 14:47:31 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id E87F628DDE; Tue,  9 Oct 2018 14:47:30 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 958E428DB0
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  9 Oct 2018 14:47:30 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726486AbeJIWEp (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 9 Oct 2018 18:04:45 -0400
Received: from mga11.intel.com ([192.55.52.93]:40091 "EHLO mga11.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1726415AbeJIWEp (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 9 Oct 2018 18:04:45 -0400
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from orsmga008.jf.intel.com ([10.7.209.65])
  by fmsmga102.fm.intel.com with ESMTP/TLS/DHE-RSA-AES256-GCM-SHA384;
 09 Oct 2018 07:47:28 -0700
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.54,360,1534834800";
   d="scan'208";a="80001248"
Received: from pnilluea-mobl1.gar.corp.intel.com (HELO
 haswell-OptiPlex-9020.gar.corp.intel.com) ([10.255.172.66])
  by orsmga008.jf.intel.com with ESMTP; 09 Oct 2018 07:47:24 -0700
From: Li Zhijian <zhijianx.li@intel.com>
To: pbonzini@redhat.com, rkrcmar@redhat.com, kvm@vger.kernel.org
Cc: Li Zhijian <zhijianx.li@intel.com>,
        Rong Chen <rong.a.chen@intel.com>,
        Philip Li <philip.li@intel.com>, 0day robot <lkp@intel.com>,
        Vitaly Kuznetsov <vkuznets@redhat.com>
Subject: [kvm-unit-tests PATCH] x86: VMX: Fix EPT MMIO access failed
Date: Tue,  9 Oct 2018 23:14:30 +0800
Message-Id: <1539098070-1885-1-git-send-email-zhijianx.li@intel.com>
X-Mailer: git-send-email 2.7.4
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

0Day robot noticed that "FAIL: EPT - MMIO access" since kernel commit
d1766202: ("x86/kvm/lapic: always disable MMIO interface in x2APIC mode")

As the changelog said, MMIO accessing is disabled in x2APIC mode
which is a common feature in modern CPUs, so disable x2APIC mode when the tests
try to access lapic MMIO accrodingly

CC: Rong Chen <rong.a.chen@intel.com>
CC: Philip Li <philip.li@intel.com>
CC: 0day robot <lkp@intel.com>
CC: Vitaly Kuznetsov <vkuznets@redhat.com>
Signed-off-by: Li Zhijian <zhijianx.li@intel.com>
---
 x86/unittests.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/unittests.cfg b/x86/unittests.cfg
index 3b21a85..883dd82 100644
--- a/x86/unittests.cfg
+++ b/x86/unittests.cfg
@@ -340,7 +340,7 @@ groups = vmx
 
 [vmx_EPT_AD_disabled]
 file = vmx.flat
-extra_params = -cpu host,+vmx -append EPT_A/D_disabled
+extra_params = -cpu host,+vmx,-x2apic -append EPT_A/D_disabled
 arch = x86_64
 groups = vmx
 
